frontend/src/__tests__/cypress/cypress/e2e/pipelines/PipelineRuns.cy.ts

```diff
@@ -170,6 +171,92 @@ describe('Pipeline runs', () => {
       });
     });
 
+    describe('table pagination', () => {
+      it('Active run table pagination', () => {
+        const mockRuns = Array.from({ length: 15 }, (_, i) =>
+          buildMockRunKF({
+            display_name: `Test active run-${i}`,
+            run_id: `run-${i}`,
+            pipeline_version_reference: {
+              pipeline_id: pipelineId,
+              pipeline_version_id: `test-version-${i}`,
+            },
+            experiment_id: `test-experiment-${i}`,
+            created_at: '2024-02-05T00:00:00Z',
+            state: RuntimeStateKF.SUCCEEDED,
+          }),
+        );
+
+        cy.intercept(
+          {
+            method: 'GET',
+            pathname: `/api/service/pipelines/${projectName}/dspa/apis/v2beta1/runs`,
+          },
+          {
+            runs: mockRuns.slice(0, 10),
+            total_size: 15,
+            next_page_token: 'page-2-token',
+          },
+        );
+
+        pipelineRunsGlobal.visit(projectName, 'active');
+        activeRunsTable.getRowByName('Test active run-0').find().should('exist');
+
+        const pagination = tablePagination.top;
+
+        // test Next button
+        pagination.findPreviousButton().should('be.disabled');
+        cy.intercept(
+          {
+            method: 'GET',
+            pathname: `/api/service/pipelines/${projectName}/dspa/apis/v2beta1/runs`,
+          },
+          {
+            runs: mockRuns.slice(10, 15),
+            total_size: 15,
+            next_page_token: 'page-2-token',
```

Comment:
There shouldn't be another page token, especially of the same value as the previous page token, when there are no more items to fetch.